### PR TITLE
`General`: Add copy button to LTI configuration

### PR DESCRIPTION
**Pull request by manuelmanso**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
1. It can be annoying to have to copy all the LTI configuration fields manually.
2. When an LTI user is initialized, they are presented with a checkbox explaining their account was created and a password was created as well. They should save the password or they will have to reset it. It's easy for users to click ok without actually reading the dialog.
3. The LTI configuration page was showing a wrong breadcrumb ("Programming Exercises")

### Description
1. Added a reusable copy-icon-button and used it in all the LTI configuration fields for which a copy button is useful.
2. Added a checkbox so that users read the LTI initialization 
3. Fixed the breadcrumb to "LTI Configuration"

### Steps for Testing
Prerequisites:
- 1 Instructor
- Login to moodle https://moodle.ase.in.tum.de/

Test copy button and breadcrumb fix

1. Log in to Artemis
2. Navigate/create online course with exercise, navigate to exercise page and then LTI configuration
3. Verify that the copy icon works as expected
4. Verify that the breadcrumb has been fixed

Test new checkbox

1. Launch an LTI exercise as a new user
2. Verify that the checkbox is present in the initialization dialog and that the user can only close the dialog when the checkbox is clicked

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|

##### Before
ltiUserInitialization

##### After

ltiUserInitAfter

##### Before
ltiConfigurationBefore

##### After
afterWithCopyButtons

### CODE REVIEW

**Review by JohannesStoehr — COMMENTED** [review]
The clone-repo-button also has a field and an additional button for copy+pasting. Maybe you can make your UI similar to this one?

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me

**Review by JohannesStoehr — DISMISSED** [review]
Code after merge looks good as well

**Review by maximiliansoelch — DISMISSED** [review]
Tested on TS1, copy buttons work and breadcrumb is fixed 👍

Regarding:
> Verify that the checkbox is present in the initialization dialog and that the user can only close the dialog when the checkbox is clicked

I managed to close the dialog without checking the box by hitting the `esc` key

File: src/main/webapp/app/shared/components/copy-icon-button/copy-icon-button.component.ts
Comment (stefanwaldhauser) [inline src/main/webapp/app/shared/components/copy-icon-button/copy-icon-button.component.ts:8]: ❌ Please write a quick unit test for this component. Should not take more than a couple of minutes

File: src/main/webapp/app/shared/components/copy-icon-button/copy-icon-button.component.ts
Comment (stefanwaldhauser) [inline src/main/webapp/app/shared/components/copy-icon-button/copy-icon-button.component.ts:9]: 💡A more specific type should be good here as onCopyFinished will be called in this way ```
onCopyFinished($event, onCopyFinishedKey)
```

**Review by stefanwaldhauser — CHANGES_REQUESTED** [review]
Code looks good and testing was fine too :) GJ. But please add a little test so the client test coverage does not drop

File: src/main/webapp/app/exercises/shared/exercise/exercise-lti-configuration.component.ts
Comment (simon-hng) [inline src/main/webapp/app/exercises/shared/exercise/exercise-lti-configuration.component.ts:61]: see if this is possible with html/css

File: src/main/webapp/i18n/de/global.json
Comment (simon-hng) [inline src/main/webapp/i18n/de/global.json:13]:
```suggestion
            "edit": "Bearbeiten",
```

**Review by akesfeden — COMMENTED** [review]
Tested on TS2 during testing session. Copy button works as expected and confirmation popup looks better now. Breadcrumb also looks good. I can approve when the client test mentioned by @stefanwaldhauser is added.

**Comment by manuelmanso** [comment]
Closing as it will be included as part of the larger https://github.com/ls1intum/Artemis/pull/5812

**Pull request closed without merging**
